ctags/json.go

```diff
@@ -43,8 +57,23 @@ func (lp *CTagsParser) Parse(name string, content []byte, typ CTagsParserType) (
 		lp.parsers[typ] = parser
 	}
 
+	deadline := time.NewTimer(parseTimeout)
+	defer deadline.Stop()
+
 	parser := lp.parsers[typ]
-	return parser.Parse(name, content)
+	recv := make(chan parseResult, 1)
+	go func() {
+		entry, err := parser.Parse(name, content)
+		recv <- parseResult{entries: entry, err: err}
+	}()
```

Comment:
I think the overhead of a new goroutine is likely fine given we know ctags is already the bottleneck. I'd be interested to know if it pops up in profiles now.

Reply:
👍 I didn't see any difference in job latency or in the CPU profiles